### ISSUE TITLE
[css-overflow-5] Focus order of generated controls

### ISSUE THREAD

**Issue by flackr** [w3c/csswg-drafts#10912]
Scroll markers #10720 and scroll buttons #10722 add focusable controls around the scrolling content. This issue is intended to cover the details of how they should be focused, which likely also implies what their relative tree order should be.

For the purposes of discussion let's consider a hypothetical DOM:
```html
<input id=before-scroller>
<div class=scroller>
  <input id=item1>
  <input id=item2>
  ...
</div>
<input id=after-scroller>
```

Styled with scroll markers and/or scroll buttons, e.g.:
```css
.scroller {
  scroll-marker-group: before | after;
}
.scroller::scroll-button(inline-start) {
  content: "<" / "Previous page";
}
.scroller::scroll-button(inline-end) {
  content: ">" / "Next page";
}
```

What should the tab order be, when you start on `#before-scroller` and tab forward? For the sake of making it easier to discuss, I think it might be helpful to define some terms:
* Buttons: All scroll-buttons, ordered by previous buttons, next buttons
* Previous buttons: All scroll-buttons navigating back, i.e. the block-start and inline-start directions in this order.
* Next buttons: All scroll-buttons navigating forward, i.e. the inline-end and block-end direction directions in this order.
* Markers: The scroll marker group, resulting in the active marker being focused.
* Before markers: The scroll-marker-group if it is `scroll-marker-group: before`.
* After markers: The scroll-marker-group if it is `scroll-marker-group: after`.
* Content: Focus goes through the content of the scroller, i.e. `#item1`, `#item2`

Options:
1. Buttons, Before Markers, Content, After Markers
2. Before Markers, Buttons, Content, After Markers
3. Before Markers, Previous buttons, Content, Next buttons, After Markers
4. Previous buttons, Before Markers, Content, After Markers, Next buttons

Some existing examples:
- [The aria apg demo](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-1-prev-next/) and @argyleink's carousel has the [previous and next buttons next to each other](https://gui-challenges.web.app/carousel/dist/) in focus order despite the latter having them visually separated.

Note:
* Once you interact with a control, we can change the next tab stop to be the targeted content similar to how following an anchor link [sets the sequential focus navigation starting point](https://html.spec.whatwg.org/#scrolling-to-a-fragment:sequential-focus-navigation-starting-point). This means that clicking on a marker or clicking on next page, and then tabbing could be defined to focus into the contents of the linked item / page regardless of the normal next item in tab order.

**Comment by flackr** [comment]
I propose we go with option 2: Before Markers, Buttons, Content, After Markers. The navigation buttons feel as if they should be directly adjacent to the content, and as suggested by aria discussions adjacent in focus order even if not visually adjacent.

One additional detail is if the scolling container itself is focusable where does it occur, my proposal would be that it is immediately before the content as this would more often result in a natural visual progression of focus indication (instead of having the container be first), i.e.
Before Markers, Buttons, Container, Content, After Markers.

**Comment by tabatkins** [comment]
I've gone ahead and reflected option 2 into the spec; we can change it if we decide differently.
